packages/eas-cli/src/credentials/ios/appstore/provisioningProfile.ts

```diff
@@ -70,6 +70,11 @@ async function addCertificateToProfileAsync(
 
   // Assign the new certificate
   profile.attributes.certificates = [cert];
+
+  // Experimentally regenerate the provisioning profile using App Store Connect API.
+  if (!profile.context.teamId && profile.context.token) {
```

Comment:
Maybe put this logic in a util so we don't duplicate the same condition in a few places.